source/class/qx/ui/form/FileSelectorButton.js

```diff
@@ -60,9 +60,19 @@
  * [MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications)
  */
 
-let InputElementIdCounter = 0;
 qx.Class.define("qx.ui.form.FileSelectorButton", {
   extend: qx.ui.form.Button,
+
+  statics: {
+    _fileInputElementIdCounter: 0,
+    _fileInputElementIdPrefix: "qxFileSelector_",
+
+    _generateElementId: function(){
+      ++this._fileInputElementIdCounter;
+      return this._fileInputElementIdPrefix + this._fileInputElementIdCounter;
+    }
+  },
+
```

Comment:
```suggestion
```
this seems overyly complex for what we try to achieve here

Reply:
I agree, could be just a static variable of course, but I really dont like to relly on the let in field scope, becouse of transpilling to ES5 in global var